### PR TITLE
Bump CAPI to v1.2.0

### PR DESCRIPTION
**Pull request by sedefsavas**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Comment by sedefsavas** [comment]
Unmanaged e2e tests passed except MachinePool (fix for it is in https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3253)

EKS test is failing.

**Comment by sedefsavas** [comment]
`The pod could not start because it could not mount the volume "aws-cred": failed to sync secret cache: timed out waiting for the condition`


/test pull-cluster-api-provider-aws-e2e-eks

**Comment by sedefsavas** [comment]
EKS tests passed: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-aws/3601/pull-cluster-api-provider-aws-e2e-eks/1547951045722771456

**Comment by sedefsavas** [comment]
/hold
 until https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3468 is merged.

File: cmd/clusterawsadm/cmd/gc/disable.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
-    http://www.apache.org/licenses/LICENSE-2.0
+	http://www.apache.org/licenses/LICENSE-2.0
```
Comment (Ankitasw) [inline cmd/clusterawsadm/cmd/gc/disable.go:8]: I see some tab changes added after rebasing GC changes, could you revert it?

File: test/e2e/shared/aws_helpers.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
-    http://www.apache.org/licenses/LICENSE-2.0
+	http://www.apache.org/licenses/LICENSE-2.0
```
Comment (Ankitasw) [inline test/e2e/shared/aws_helpers.go:11]: are these changes intentional?

File: test/e2e/data/e2e_eks_conf.yaml [hunk 3]
```diff
@@ -109,15 +88,6 @@ providers:
     files:
       - sourcePath: "./shared/v1alpha4/metadata.yaml"
     versions:
-      - name: v0.3.23 # latest published release in the v1alpha3 series; this is used for v1alpha3 --> v1beta1 clusterctl upgrades test only.
```
Comment (Ankitasw) [inline test/e2e/data/e2e_eks_conf.yaml:112]: Are we deprecating v1alpha3 upgrade tests? If yes, shall we also think of removing those from testgrid?

File: config/crd/bases/infrastructure.cluster.x-k8s.io_awsclustercontrolleridentities.yaml [hunk 4]
```diff
@@ -104,6 +104,7 @@ spec:
                           "value". The requirements are ANDed.
                         type: object
                     type: object
+                    x-kubernetes-map-type: atomic
```
Comment (Ankitasw) [inline config/crd/bases/infrastructure.cluster.x-k8s.io_awsclustercontrolleridentities.yaml:107]: Is this SSA related parameter?

**Comment by Ankitasw** [comment]
The E2E tests should pass now.
/test pull-cluster-api-provider-aws-e2e

File: cmd/clusterawsadm/cmd/gc/disable.go (hunk 1 above)
Comment (sedefsavas) [reply cmd/clusterawsadm/cmd/gc/disable.go:8]: These are the new files, added with GC PRs, linter was failing. Boilerplate was not matching with the existing ones.

File: test/e2e/shared/aws_helpers.go (hunk 2 above)
Comment (sedefsavas) [reply test/e2e/shared/aws_helpers.go:11]: These are the new files, added with GC PRs, linter was failing. Boilerplate was not matching with the existing ones.

File: test/e2e/data/e2e_eks_conf.yaml (hunk 3 above)
Comment (sedefsavas) [reply test/e2e/data/e2e_eks_conf.yaml:112]: Yes, after this is merged, we will need to remove them.

File: config/crd/bases/infrastructure.cluster.x-k8s.io_awsclustercontrolleridentities.yaml (hunk 4 above)
Comment (sedefsavas) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_awsclustercontrolleridentities.yaml:107]: These are auto generated files, this change is due to dependency bump.

**Comment by sedefsavas** [comment]
GPU test might still fail, I might need to rebase to get the GPU fix in.

**Comment by sedefsavas** [comment]
Tests passed.

/hold cancel
